graphbolt/src/random.cc

```diff
@@ -0,0 +1,61 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file random.cc
+ * @brief Random Engine.
+ */
+
+#include "./random.h"
+
+#include <torch/torch.h>
+
+namespace graphbolt {
+
+namespace {
+
+// Get a unique integer ID representing this thread.
+inline uint32_t GetThreadId() {
+  static int num_threads = 0;
+  static std::mutex mutex;
+  static thread_local int id = -1;
+
+  if (id == -1) {
+    std::lock_guard<std::mutex> guard(mutex);
+    id = num_threads;
+    num_threads++;
+  }
+  return id;
+}
+
+};  // namespace
+
+std::optional<uint64_t> RandomEngine::manual_seed;
```

Comment:
Is this repeat with static variable in header file?

Reply:
This is initialization of static variable which declared in the header file.